avr-hal-generic/src/spi.rs

```diff
@@ -75,9 +75,29 @@ macro_rules! impl_spi {
                 sclk: $sclkmod:ident::$SCLK:ident,
                 mosi: $mosimod:ident::$MOSI:ident,
                 miso: $misomod:ident::$MISO:ident,
+                cs: $csmod:ident::$CS:ident,
             }
         }
     ) => {
+
+        /// Wrapper for the CS pin
+        ///
+        /// Used to contain the chip-select pin during operation to prevent its mode from being
+        /// changed from Output.  Implements
+        /// [`OutputPin`](https://docs.rs/embedded-hal/latest/embedded_hal/digital/v2/trait.OutputPin.html)
+        /// so that it can still be controlled.
+        pub struct ChipSelectPin($csmod::$CS<$crate::port::mode::Output>);
+
+        impl $crate::hal::digital::v2::OutputPin for ChipSelectPin {
+            type Error = <$csmod::$CS<$crate::port::mode::Output> as $crate::hal::digital::v2::OutputPin>::Error;
```

Comment:
Rust syntax at its finest :D  I would just use the concrete type though, should be `$crate::void::Void` if I'm not mistaken.

Reply:
Oh sweet, that makes it much easier.